crates/router/src/connector/bambora/transformers.rs

```diff
@@ -0,0 +1,176 @@
+use masking::{PeekInterface};
+use serde::{Deserialize, Serialize};
+use crate::{core::errors,types::{self,api, storage::enums}};
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct BamboraPaymentsRequest {
+    amount: i64,
+    payment_method: String,
+    card: Card
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BamboraPaymentsRequest  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        let payment_method_data = item.request.payment_method_data.clone();
+        let payment_method =
+            match payment_method_data {
+                api::PaymentMethod::Card(ref _item) => String::from("card"),
+                _ => todo!()
+            };
+        let card =
+            match payment_method_data {
+                api::PaymentMethod::Card(ref item) => {
+                    Card {
+                        name: item.card_holder_name.peek().clone(),
+                        number: item.card_number.peek().clone(),
+                        expiry_month: item.card_exp_month.peek().clone(),
+                        expiry_year: item.card_exp_year.peek().clone(),
+                        cvd: item.card_cvc.peek().clone()
+                    }
+                }
+                _ => todo!()
+            };
+        Ok(Self {
+            amount: item.request.amount,
+            payment_method: payment_method,
+            card,
+        })
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// Auth Struct
+pub struct BamboraAuthType {
+    pub(super) api_key: String
+}
+
+impl TryFrom<&types::ConnectorAuthType> for BamboraAuthType  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::HeaderKey { api_key } = _auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+    }
+}
+// PaymentsResponse
+//TODO: Append the remaining status flags
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum BamboraPaymentStatus {
+    #[default]
+    ZERO,
+    ONE,
+}
+
+impl From<BamboraPaymentStatus> for enums::AttemptStatus {
+    fn from(item: BamboraPaymentStatus) -> Self {
+        match item {
+            BamboraPaymentStatus::ONE => Self::Charged,
```

Comment:
what if the payment status is pending?

Reply:
Didn't find any info about pending in documentation